File: backend/tiggle-root/tiggle-auth-gateway/src/main/java/com/side/tiggle/authgateway/oauth/OAuth2FailureHandler.java
package com.side.tiggle.authgateway.oauth;

import org.springframework.security.core.AuthenticationException;
import org.springframework.security.web.server.WebFilterExchange;
import org.springframework.security.web.server.authentication.ServerAuthenticationFailureHandler;
import org.springframework.stereotype.Component;
import reactor.core.publisher.Mono;

/**
 * TODO : OAuth2 인증 실패에 대한 처리를 추가한다
 */
@Component
public class OAuth2FailureHandler implements ServerAuthenticationFailureHandler {
    @Override
    public Mono<Void> onAuthenticationFailure(WebFilterExchange webFilterExchange, AuthenticationException exception) {
        return null;
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/global/auth/oauth2/OAuth2FailureHandler.kt
package com.side.tiggle.global.auth.oauth2

import com.side.tiggle.global.exception.AuthException
import com.side.tiggle.global.exception.error.GlobalErrorCode
import org.springframework.security.core.AuthenticationException
import org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler
import org.springframework.stereotype.Component
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse

@Component
class OAuth2FailureHandler: SimpleUrlAuthenticationFailureHandler() {

    override fun onAuthenticationFailure(request: HttpServletRequest?, response: HttpServletResponse?, exception: AuthenticationException?) {
        logger.error("OAuth2 authentication failed: ${exception?.message}", exception)
        throw AuthException(GlobalErrorCode.OAUTH2_AUTHENTICATION_FAILED)
    }
}